webpack/scenes/RedHatRepositories/RedHatRepositoriesPage.js

```diff
@@ -0,0 +1,118 @@
+/* eslint-disable import/no-extraneous-dependencies */
+/* eslint import/no-unresolved: [2, { ignore: [foremanReact/*] }] */
+/* eslint-disable import/no-unresolved */
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Grid, Row, Col } from 'react-bootstrap';
+import { Button } from 'patternfly-react';
+import { translate as __ } from 'foremanReact/common/I18n';
+import PermissionDenied from 'foremanReact/components/PermissionDenied';
+import { LoadingState } from '../../move_to_pf/LoadingState';
+import { createEnabledRepoParams } from '../../redux/actions/RedHatRepositories/enabled';
+import SearchBar from './components/SearchBar';
+import RecommendedRepositorySetsToggler from './components/RecommendedRepositorySetsToggler';
+import { getSetsComponent, getEnabledComponent } from './helpers';
+import api from '../../services/api';
+
+class RedHatRepositoriesPage extends Component {
+  componentDidMount() {
+    this.loadData();
+  }
+
+  loadData() {
+    this.props.loadEnabledRepos();
+    this.props.loadRepositorySets({ search: { filters: ['rpm'] } });
+  }
+
+  render() {
+    const { enabledRepositories, repositorySets } = this.props;
+    const { repoParams } = createEnabledRepoParams(enabledRepositories);
+
+    if (repositorySets.missingPermissions && repositorySets.missingPermissions.length > 0) {
+      return <PermissionDenied missingPermissions={repositorySets.missingPermissions} />;
+    }
+    if (enabledRepositories.missingPermissions
+      && enabledRepositories.missingPermissions.length) {
+      return <PermissionDenied missingPermissions={enabledRepositories.missingPermissions} />;
+    }
+
+    return (
+      <Grid id="redhatRepositoriesPage" bsClass="container-fluid">
+        <h1>{__('Red Hat Repositories')}</h1>
+        <Row className="toolbar-pf">
+          <Col sm={12}>
+            <SearchBar />
+          </Col>
+        </Row>
+
+        <Row className="row-eq-height">
+          <Col sm={6} className="available-repositories-container">
+            <div className="available-repositories-header">
+              <h2>{__('Available Repositories')}</h2>
+              <RecommendedRepositorySetsToggler
+                enabled={repositorySets.recommended}
+                onChange={value => this.props.updateRecommendedRepositorySets(value)}
+                className="recommended-repositories-toggler"
+              />
+            </div>
+            <LoadingState loading={repositorySets.loading} loadingText={__('Loading')}>
+              {getSetsComponent(
+                repositorySets,
+                (pagination) => {
```

Comment:
would it make sense to move it into a class handler? brackets can be removed too.

Reply:
@laviro Yes, I think this entire page could use a refactor.  But maybe in another PR.

Reply:
Yeah, I agree..
My first impression was that you wrote it from scratch.

Reply:
Thanks, sorry, my first impression was that you wrote it from scratch.